code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -37,17 +46,32 @@ var StageProgress = React.createClass({
         }
       }
 
-      return ([
-        <div className={outerClass}>
-          <a
-            href={level.url}
-            className={innerClass + ' level-' + level.id}
-            style={dotStyle}>
-            {level.title}
-          </a>
-        </div>,
-        ' '
-      ]);
+      if (this.props.saveAnswersFirst) {
+        dotStyle.cursor = 'pointer';
+        return ([
+          <div className={outerClass}>
+            <span
+              onClick={()=>{this.dotClicked(level.url);}}
+              className={innerClass + ' level-' + level.id}
+              style={dotStyle}>
+                {level.title}
+            </span>
+          </div>,
+          ' '
+        ]);
+      } else {
+        return ([
+          <div className={outerClass}>
+            <a
+              href={level.url}
+              className={innerClass + ' level-' + level.id}
+              style={dotStyle}>
+                {level.title}
+            </a>
+          </div>,
+          ' '
+        ]);
+      }
```

Comment:
Can we combine this behavior so there's only one way to render a progress dot?  Either keep `<a>` and add `e.preventDefault()` to the `saveAnswersFirst` handler, or use `<span>` and add a default handler for normal navigation (I slightly prefer the first option).